src/main/resources/data/computercraft/lua/rom/programs/poster.lua

```diff
@@ -27,68 +37,137 @@ if args[1] == "stop" then
   return
 end
 
+local status = printer.status()
+if status == "busy" then
+  error("Printer is busy", 0)
+end
+
+-- Convert string to number and validate it.
+local function getNum(str)
+  local num = tonumber(str)
+  if not num or math.floor(num) < 1 then
+    return
+  end
+  return math.floor(num)
+end
+
+-- Get count.
 local count = 1
-if #args == 2 then
-  count = tonumber(args[2])
-  if not count or math.floor(count) < 1 then
+if #args > 1 then
+  count = getNum(args[2])
+  if not count then
     printUsage()
     return
   end
-  count = math.floor(count)
 end
 
--- Load the 2dj file
+-- Get index, if given.
+local index = { all = true, from = nil, to = nil }
+if #args >= 3 then
+  local num1, num2 = getNum(args[3]), getNum(args[4])
+  if not num1 or not num2 then
+    printUsage()
+    return
+  end
```

Comment:
Have you tested this works the way you wanted?

Unless I'm reading this wrong, it looks like this will printUsage if args[4] is nil (but args[3] is valid), in which case specifying only an index would not work.

Reply:
The idea behind this code is that if either of the given `args` is not a valid number, the usage will be printed, as something else that we can not accept got provided. `getNum` does the check for us already.

If nothing was provided in `args[3]` and `args[4]`, we expect that the user wants the whole file to be printed.

Reply:
If only args[3] is provided, e.g. `poster test.2dja 1 8`, based on your help text this should print only the 8th page. But based on this code, it will error out since args[4] is nil.

Reply:
Oh I see. Yeah I'll fix that.